### DISCUSSION TITLE
How is state maintained across browsers for same code-server Instance ?

### DISCUSSION THREAD

**Discussion by abhay-ranawat** [coder/code-server#4185] [Q&A]
Before v1.60 there was a folder named `state` in `~/.local/share/code-server/User/state`, but after the update the folder is missing, and I have to perform the same actions across browsers as if it was a new installation.
So, how can I maintain the same stated across browsers from server side.

**Comment by jsjoeio** [comment]
Good question! @TeffenEllis any ideas on what we might have changed here? (Or rather VS Code might have changed)

**Reply by abhay-ranawat** [reply]
Hey @jsjoeio @TeffenEllis any update ?

**Reply by jsjoeio** [reply]
No updates on mine end. I'm sure you'll hear back soon (today or next week).

If you can provide reproduction steps or logs or something, I can try looking into it myself, but Teffen will have the most context.

**Reply by ghost** [reply]
I have the same issue. It maintains states at it was previously on the device which is used but no more inter-device state synchronization.

**Reply by abhay-ranawat** [reply]
@jsjoeio what kind of logs, afaik it's storing the state in local browser.

**Reply by abhay-ranawat** [reply]
@jsjoeio To reproduce open your code server in incognito tab, and have git lens installed

**Comment by GirlBossRush** [comment]
I’m guessing this is due to Microsoft’s migration towards Settings Sync for UI state in addition to the typical editor preferences. I’m not yet certain how much they’ve changed, but I suspect we can resolve this behavior in our next release.

**Comment by DmitriGoloubentsev** [comment]
Hi all, many thanks to topic starter. 

I also would like to maintain state across browsers and this issue causes lots' of problems to me.

Looking at the source I found this :
`scope === StorageScope.GLOBAL ? 'global' : this.payload.id;` it is in workbench.web.api.js.map
Perhaps it's related to where the state is saved?

If anyone has a temporary workaround, I would be glad to know about it!

Thanks!

**Reply by jsjoeio** [reply]
Would you or @abhay-ranawat be able to open a bug report with minimal repro steps? Then we can track looking into and fixing this!

**Reply by abhay-ranawat** [reply]
@jsjoeio https://github.com/cdr/code-server/issues/4212
